index.js

```diff
@@ -316,6 +315,94 @@ const legacyScriptingRunner = (Zap, zcli, input) => {
     return _.template(templateString, options)(values);
   };
 
+  const ensureIsType = (result, type) => {
+    if (type) {
+      if (type.startsWith('array-')) {
+        if (Array.isArray(result)) {
+          return result;
+        } else if (result && typeof result === 'object') {
+          if (type === 'array-wrap') {
+            return [result];
+          } else {
+            // Find the first array in the response
+            for (const k in result) {
+              const value = result[k];
+              if (Array.isArray(value)) {
+                return value;
+              }
+            }
+          }
+        }
+        throw new Error('JSON results array could not be located.');
+      } else if (type.startsWith('object-')) {
+        if (_.isPlainObject(result)) {
+          return result;
+        } else if (
+          Array.isArray(result) &&
+          result.length > 0 &&
+          _.isPlainObject(result[0])
+        ) {
+          // Used by auth test and auth label
+          return result[0];
+        }
+        throw new Error('JSON results object could not be located.');
+      }
+    }
+    return result;
+  };
+
+  const isObject = (object) => {
+    return !Array.isArray(object) && typeof object === 'object';
+  };
+
+  const handleLegacyParams = (data) => {
+    // see handle_legacy_params in the python backend
+    if (!isObject(data)) {
+      return data;
+    }
+    var params = {};
+    for (var i in data) {
+      if (isObject(data[i])) {
+        var param = [];
+        for (var j in data[i]) {
+          param.push(j + ',' + data[i][j]);
+        }
+        params[i] = param.join('|');
+      } else if (Array.isArray(data[i])) {
+        params[i] = data[i].join();
+      }
+    }
+    return params;
+  };
+
+  const textifyList = (data) => {
+    // see textify_list in the python backend
+    if (!Array.isArray(data)){
+      return data;
+    }
+    var out = '';
+    const allObj = data.every(isObject);
+    const sep = allObj ? '\n' : ',';
+
+    for (var i in data) {
+      if (isObject(data[i])) {
+        for (var key in data[i]) {
+          out += key + ': ' + data[i][key] + sep;
+        }
+      } else if (Array.isArray(data[i])) {
+        out += textifyList(data[i]) + sep;
+      } else if (data[i] == null) {
```

Comment:
`if line is None` is sort of a tough line to translate into js. The best thing to put here really depends on what sort of data you think `data[i]` could be. I'd take a look through [these tables](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comparison_Operators#Equality_operators) and make sure this will evaluate as you expect.

You might already know all this, but just in case I figured it was worth flagging!

Reply:
Good catch, I wasn't aware that `undefined == null`. I've changed this in a20c0f9e35683eeeaaf14d6512f4e1583f99861b to use the strict equality operator, which I believe is the behaviour we want:
```
> x = {1:null}
{ '1': null }
> x[1] === null
true
> x[2] === null
false
```

Reply:
welcome to the wide world of JS equality! It mostly make sense but `==` has some very [odd behaviors](https://github.com/denysdovhan/wtfjs#table-of-contents)